File: src/main/java/com/example/message/notifier/MessageNotifier.java
package com.example.message.notifier;

import com.example.message.domain.Message;
import org.springframework.stereotype.Component;

import java.util.ArrayList;
import java.util.List;

@Component
public class MessageNotifier implements Notifier {

    public MessageNotifier() {
        notifiers = new ArrayList<>();
        notifiers.add(new FirebaseNotifier());
    }

    @Override
    public void notify(Message message){
        for (Notifier notifier : notifiers) {
            notifier.notify(message);
        }
    }

    private final List<Notifier> notifiers;
}

File: src/main/java/com/example/message/notifier/MessageNotifier.kt
package com.example.message.notifier

import com.example.message.domain.Message
import org.springframework.stereotype.Component

@Component
class MessageNotifier : Notifier {

    override fun notify(message: Message) {
        notifiers.forEach {
            it.notify(message)
        }
    }

    private val notifiers = listOf<Notifier>(
            FirebaseNotifier()
    )
}